### ISSUE TITLE
We need to figure out how we're going to do special attacks

### ISSUE THREAD

**Issue by torch2424** [torch2424/Fight#82]
Emanual has been making beautiful AOE attacks, and all kinds of specials, but How do we manage this in gameplay. Do we need an mp bar? How will MP regenerate?

I am against adding an mp section in our hud, as it is already taking up alot of space, and I suggest that you just automatically use one after you attack so many times, and depending on your level, you do a different special.

You could flash yellow to indicate it's a special attack, and make a sounds, kinda like tony hawk

**Comment by torch2424** [comment]
Especially because of #61 , that's going to need it's own dedicated button as well, and that's even more screen space (On mobile) and more even complicated to play

**Comment by emanuelpartida** [comment]
I think there should be mp and some enemy should drop potions

**Comment by Heebe** [comment]
I agree with Emanuel, and with your concerns about taking up space, we could just change the HUD around a little, like so
![concept](https://cloud.githubusercontent.com/assets/7026914/8553227/c5e37eea-2496-11e5-98d1-e6c3d85490d3.png)

**Comment by Heebe** [comment]
and if you want some extra space on top of that, we abbreviate "level" to just "lvl"

**Comment by torch2424** [comment]
I like that idea of moving the hud up there, that actually makes alot more sense thank you @Heebe 

@emanuelpartida @Heebe One thing about that though that I've discussed with daniel before, but he and I agreed a while ago that it isn't one thing that would be good to do for the following reasons

A: It would make every playthrough more luck based. Because if I have enemies drop items at random then the amount of all-powerful special attacks increase. therefore if we add any type of multiplayer or leaderboards, that would be a bit more biased.

B: As I said before it would take up screen space. Not just in the hud, but Imagine mobile. We already have planned a d-pad (analog stick) and a punch button, and a dodge button. I feel like I would have to make the other two buttons smaller, which would make them harder to press, or keep them all big size, and you lose track of our character even more cause you cant see him, which has already been becoming an increasing problem. Also, 3 buttons on the right hand side of the screen is an awkward number.

C: Having items on the map would clutter things up even more with our permanance and kevin's object. I feel like we're already pushing boundaries of this game (Which is a good thing) and we need to keep in mind how far we are pushing it

D: I could go on and on with other problems with this but I'm at work, and I wanted to get my say in as you guys kept pushing for this

**Comment by Heebe** [comment]
Alright, I have an idea.

-Mobile-
Buttons:
Analog, Punch Button, Dodge Button, Start, Select.

Analog = Movement
Punch Button = Punch
Dodge Button = Dodge
Start = menu to exit game, restart, or options button.
Select = Pulls up Shop/Powers Menu

The Shop/Powers menu allows you to purchase special attacks with score, or if we decide, with real money. But don't completely lock any powers behind real money. This menu also allows you to bind a certain special power that you have as "usable".

When a person playing wants to use that move that they selected, all they need to do is tap the middle of their screen.

-Desktop Computer-

Same as above but we could give the player more options to bind powers to keys on their keyboard, also, add keymapping.

_The MP issue_
I like the idea of a flash of yellow signaling the player that they have the special power readily available so we'll keep that. We won't visibly use "MP" though. What I mean is that what if we do have random drops of "Orbs" 1/30th chance to drop, collect 3 - 5 and your character gets the yellow flash showing the power is available. The Orb will be on the formost layer "just under the special attack" so you will always be able to see it. The orbs will disappear after 5 - 10 seconds as well.

_The Multiplayer Issue_

I do not think it would add bias to multiplayer. It could add strategy even, with 2 players having 2 different specials set, one player could have their partner collect the orbs so that they could use their special if they have a good one.

If a player in multiplayer accesses the Shop/Power menu, on the other person's screen the character will be shown guarding up or something, with them being immune for a small amount of time. Enough to let the person select a different special but to know they have a limited amount of time to get back into action.

This issue is a work in progress but still.

_No issue with Leaderboard_
It's still a skill based game, with this random drop or not.

**Comment by torch2424** [comment]
@Heebe 
@emanuelpartida 

Having a select button doesn't fix the issue of having too many buttons. Honestly, I forgot we would even need a start button for mobile. I think having 4 buttons on a mobile game should be max, that's already a lot of screen space.

I don't think we should use a shop system for our special attacks, I feel like score could be used for you to unlock special attacks which would be cool, but stoping the flow of the game for them is a bad idea in my opinion. I feel like it works in counterstrike, because It's alot more tactical, ours is more all-out action, stopping that would be a HUGE drop off point for our players

Using real money to unlock powers I feel like is kinda too pay to win. I think we should use in app purchases to unlock new maps and things. maybe even new players with different stats. but mostly aesthetic things. Like for example, our current uy the default one could be 1 health, 1 seed, 1 attack. But you can buy a character for .5 health, .5 speed, 2 attack, and he's purple. I think that'd be better as it gives players a fair advantage depending on their play style, now that we've added things like health regen i think it'd be the way to go. We could also make special maps or bosses purchaseable, and we could definitely make dlc purchaseable.

Honestly, for our business plan with the game, I think it should be offer a demo of just the core gameplay, and you can't unlock anything (kind of like a demo, but you can play forever). But through in app purchases, or buying the game, you get the ability to unlock aesthetic games and stuff. I'll document my ideas on this in this issue: https://github.com/torch2424/Punch-Legacy/issues/104

And for tapping the middle of the screen, that is not a good idea in my opinion, people are going to be tapping the screen like crazy, with 4 on screen buttons as it is, our real estate is lacking, and the middle of the screen is just a slight mistap. That is why I suggested that the attack button just turn a different color or something and you use a special attack

I'm glad we agreed on flashing yellow. But the permanence has our map already intensely cluttered. The only way we could have item drops without the game looking just too much is if we get rid of some of the permenance, and having the enemies delete themselves after a specified time.

I like the idea of two different players having specials. in the issue I referenced, I left a note of that. But again I made an argument about the shop menu, and having your buddy guard stops the flow of the game, for not only you, but makes a multiplayer experience feel singular.

And I still think drops is an issue with the leaderboard. If drops still made a game completely skill based. How come in all of competitive smash play, they don't allow smash balls or any health drops, or item drops? Even nintendo realized that drops make the game for fun, compared to for glory. When people start getting competitive, things like item drops become a huge concern

**Comment by torch2424** [comment]
Also, I removed this from the milestone as this has been quite a talk for a while now, and I do not think it will be solved soon

**Comment by torch2424** [comment]
@Heebe I'm looking forward to your response

**Comment by emanuelpartida** [comment]
I think we should use an mp system that regenerates like health but a lot slower. It should activate it when both attack and dodge are pressed.

**Comment by torch2424** [comment]
That's a good idea for an mp system, but I think pressing space and ctrl is an awkward combination, and on mobile it's be even harder

**Comment by torch2424** [comment]
@Heebe @emanuelpartida after some play testing, it was suggested to me, that we have powerups that drop such as double damage, or double speed, that last for about 30 seconds to a minute until they go away. We could do that, or have specail attacks automatically activate once you pick them up

**Comment by Heebe** [comment]
Doesn't that drop system completely contradict what you were saying about drops unbalancing the game though?

**Comment by Heebe** [comment]
But yeah I like the MP system too.

**Comment by Heebe** [comment]
I think the button mapping as it stands right now is a little bit awkward too. It's a bit hard to play with ctrl but is playable with the wasd, r, and space combo, still a bit awkward though.

**Comment by torch2424** [comment]
@Heebe yes it does, but it was just a suggestion. I don't know if you've agreed or disagreed with drop unbalancing the game yet.

I'm glad we decided on an mp system, but if we use that system, how do we do unleashing the attacks?

And I made the r and ctrl off of the top of my head, that can be changed, just create an issue for it @Heebe

**Comment by Heebe** [comment]
Yeah I just don't see drops being a good idea as of yet. I'm still thinking on how we should unleash the attacks.

**Comment by torch2424** [comment]
Okay awesome, WE ARE AGREED NO DROPS AS IT CAN LEAD TO UNBALANCED GAMEPLAY, SEE ABOVE FOR MORE

**Comment by torch2424** [comment]
@Heebe I remember someone suggested holding the attack button, we could do that, and like we can have them flash so many times, and if they do unleash the attack

**Comment by Heebe** [comment]
Holding the attack button is already taken for continuous attacking, and we
should keep the continuous attacking because that's something I've had been
requested to by numerous people.

**Comment by torch2424** [comment]
Cool. Well then @Heebe I think honestly our best bet for this, since we cant really execute special attack, maybe this isn't the game to have it. We'd have to sacrifice dodging for them at this point. And honestly dodging is preety cool. I think we should simply just go with the drops that give you double damage for a time period. Yeah it's unbalanced, but if anything, we can have a for glory mode without them. And I can code a backend with two different leaderboards for each respectively, if a competitive leaderboard scene does arrive

**Comment by torch2424** [comment]
I had multiple people suggest drops with double damage and stuff, wouldn't be too hard to implement, but we'd need to you to create drop icons for this. As I feel like we have the cake, and now we just need the icing

**Comment by emanuelpartida** [comment]
Special attacks are much cooler though!!!! Goku isn't goku with out his Kamehameha!!! Punch legacy needs Fire Ball Shots and Earth spikes!!!!

**Comment by emanuelpartida** [comment]
You could always use V to dodge and it would make it easier to press Space and V together.

**Comment by emanuelpartida** [comment]
if you press V and space together you shoot fire and if you hold it your character charges and does earth spike

**Comment by torch2424** [comment]
@emanuelpartida You make a good point, goku isnt as cool without his kamehameha hahahaha :')

But yo, yeah that's work on desktop, sure. But you gotta think cross platform, would it work on a phone? or a PSP, or DS?

Mainly our goal is to keep in mind  mobile

**Comment by emanuelpartida** [comment]
i think it would work on mobile you would stop running and have to use your special move and a DS and PSP you could press L and R or Hold it. I just want to shoot fire and summon spikes from the ground.

**Comment by emanuelpartida** [comment]
While holding your special attack the screen should shake making it seem like your gathering energy or something.

**Comment by torch2424** [comment]
@emanuelpartida But how would it work on mobile? Some phones don't even have any physical buttons. Which means we have to put buttons on the screen. And as of right now, we already have 4 buttons, with a hud. A fifth buton would just be too much, and block gameplay viewing.

But yeah I could easily do the screenshaking thing

**Comment by emanuelpartida** [comment]
![buttons](https://cloud.githubusercontent.com/assets/13184539/8945891/8363b692-3540-11e5-9897-8b52e9054af2.png)
what if the buttons were like that?

**Comment by emanuelpartida** [comment]
![buttons](https://cloud.githubusercontent.com/assets/13184539/8945914/b845a712-3540-11e5-9302-9b297ee91af2.png)
what if the buttons were like this?

**Comment by Heebe** [comment]
Those buttons are waaaaaaayyyyy too small

**Comment by emanuelpartida** [comment]
Make the bigger but same layout

**Comment by Heebe** [comment]
Yeah I like that idea, I'll sketch it and refine it when I get home to show
how I invision it.

**Comment by emanuelpartida** [comment]
Fosho!

**Comment by Heebe** [comment]
![controls](https://cloud.githubusercontent.com/assets/7026914/8947871/57904e1c-3555-11e5-9004-75790f5415a9.png)
P for punch (can be A for attack),
D for dodge,
S for special.

Buttons can be transparent to not block too much of the screen.

It was tough making everything the perfect proportion in paint but this is the jist of it. The bar at the top is there for no reason really, just made it easier to draw.

**Comment by torch2424** [comment]
Alright cool we Can go with this design, I'm glad we finally got this figured out. one thing I was hoping for was bigger buttons to help against mis clicks, and larger hands,. But this design may work out. @Heebe I'll need you to make these buttons through for the art. Dont worry about a pressed state, I can handle that

**Comment by torch2424** [comment]
Also, might I suggest an analog stick instead of dpad? And I'll move this into the controls issue in a bit.

**Comment by torch2424** [comment]
So the final ultimatum:

Mp regenerates over time like health

We will add a 5th button (8th if dpad) for unleashing special attacks

special attacks will be earned/usable depending on the player's current level range. One per level range

character will flash when special is ready

No drops as this clutters screen with already existing permenance, and unbalances gameplay

Is everyone agreed?

**Comment by Heebe** [comment]
"one thing I was hoping for was bigger buttons to help against mis clicks, and larger hands", In regards to that we can just see how big we can make them without it being too intrusive.
For the part about analog vs d-pad, I was thinking about an analog stick too but it was just a little too hard to draw in paint plus the arrows were already a pre-set shape so it was just easier to draw for the example.

**Comment by Heebe** [comment]
> No drops as this clutters screen with already existing permenance, and unbalances gameplay

I thought we were going to do the "double speed" and etc. power-ups as well?

**Comment by Heebe** [comment]
-Mp regenerates over time like health
I like.
-We will add a 5th button (8th if dpad) for unleashing special attacks
Yeah.
-special attacks will be earned/usable depending on the player's current level range. One per level range
I'm not understanding this.
-character will flash when special is ready
Again, I don't understand this. When is it going to flash? We know that it's going to work off of MP.

**Comment by torch2424** [comment]
@heebe yes exactly we can re size, and yes when you make the images, you need to make a larger circle, and a smaller circle.

For drops emanual stated: "Special attacks are much cooler though!!!!"

therefore he said that'd special attacks are the better idea. Drops and specials attacks would be overkill in my opinion

**Comment by PartidaE** [comment]
Over kill would be a cool thing too because imagine all that power!!!!! Muhahaha

**Comment by Heebe** [comment]
Yeah I agree with Emanuel! Both wouldn't be overkill, it would be perfect. It would be FAST, ACTION PACKED, NON. STOP. ACTION!!! But really, it would be really fun and it'd keep players super entertained.

**Comment by torch2424** [comment]
@Heebe  @emanuelpartida Are you completely positive? You've thought this through?

I'm being outvoted, and I think I'm just being too picky on this whole thing. How about this. lets just start over, I think I kinda influenced everything oddly. You and emanual can come up with an idea wether it be more buttons, more mana, orbs, specials drops or whatever. And then come back to me, and I'll code it up?

sound good? :)

**Comment by Heebe** [comment]
![controls](https://cloud.githubusercontent.com/assets/7026914/8947871/57904e1c-3555-11e5-9004-75790f5415a9.png)
Well these two things are good for sure and we all agree on them.
-Mp regenerates over time like health.
-We will add a 5th button.
We suggest power-up drops like increased speed or increased damage, along with having special powers, hence the fifth button (for mobile). We use the type of control scheme for mobile as pictured above.

We need to know when special attacks can be used. We already know they would work off of MP. I was thinking we use the different special for different attack ranges thing, and make each special more powerful within a new range, but make it use more MP as well to balance it.

Also, suggested by @emanuelpartida, a power-up drop can be a transformation were punch-guy turns into a large fighting monster for a short time, but it must be a RARE drop.

**Comment by Heebe** [comment]
Also, we implement the suggestion from @torch2424 for a "for glory" mode with separate leaderboards for the more competitive fanbase. The mode will have no drops, and (no mp?).

**Comment by torch2424** [comment]
@Heebe The for glory mode was simply an example I was making with the drops. I apologize if I made it seem like I wanted to go through with it. If anything, we can keep For Glory on the backburner "If a competitive leaderboard scene does arrive" this will probably be more pertinent if multiplayer joins as well. But for now, since the direction you guys want to go, the game is gonna be more geared for randomness and strategy, compared to skill. It'll be a bit difficult for people to take it as a more skillgul game, which isn't a bad thing though.

Also, I brought this up with Daniel, one thing he said we should keep in mind is that the game is called "Super Punch Legacy", he emphazied that the game title implies the game revolves around punchng. So our attacks and specials should reflect that

**Comment by torch2424** [comment]
Also, one thing about your image, I thought you should see this to explain why I was worried about the amount of buttons

![2015-07-30](https://cloud.githubusercontent.com/assets/1448289/9002189/bc1d8a22-3715-11e5-926f-7e106e6d6d1b.jpg)

I have a 4.5 inch screen. Your picture is only slightly smaller than my device in comparison, and to be honest, the image makes my device look even a little bigger than it does in person, as it's a little bit closer to the camera than my laptop screen. Even though phones are a bit bigger now. Iphone 5's have about my screen size, and so do a good amount of android devices. Alot of people cant afford flagships, and tend to get devices like mine. Especially preeteens or young kids, who will more than likely be our mobile market

**Comment by Heebe** [comment]
I still think the button orientation would work. It would crowd to much of the screen especially if the buttons are transparent. Most likely the S button would be used more tactically so it won't be pressed as much and making the real estate easy to handle. I just did a 4.5 inch screen measurement using my own phone, and although the screen is small, it wont take up as much space as you think and will still allow for comfortable button presses. If you can think of an idea for button placement that includes a special button though, go right ahead.

**Comment by Heebe** [comment]
Daniel has a point with specials needing to be "punch oriented", noted, and referred to @emanuelpartida

**Comment by Heebe** [comment]
And ok, @torch2424 but if the need arises we should just keep a "vanilla" mode like that in the back of our minds.

**Comment by torch2424** [comment]
@Heebe I understand that the buttons would be an alright size. But I wanted to have them larger, as we are getting close to an snes button config:

![unnamed](https://cloud.githubusercontent.com/assets/1448289/9012741/1b117452-376c-11e5-9007-37cd3f2e6efc.jpg)

Like it was known from the beginning the buttons would be semi transparent. One thing I'd like to ask, have you ever tried to play streetfighter 2 turbo mode on an snes emulator? It's almost no possbile, there's so many misclicks, your fighting more of the controller than the actual fighter. Our game might even be more fast paced than that. so whatever button configuration we chooses, should be a bit larger than comfortable, even slightly larger than the picture above I'd think. Also, notice how much of the screen is actually clearly visibile. A little less than half right? butcause the dpad and the extra third and forth buttons do that. And dont substract L and r, that's where our hud is going, which wont even be transparent.

I'm not in favor of not having a button, I just want to show you my thought process on these kinds of decisions.

And col. I'll be working on the game all day today. I wont be doing special attacks till next week. As I still want you an Emanuel to figure this out.

**Comment by Heebe** [comment]
I understand that, but super punch legacy isn't a 2d fighter like street fighter or mortal kombat. It's a fluid almost-topdown arcade style fighter. 2 different genres. I believe our button scheme would work with ours, but of course not with a game like that. Even if our game is fast paced.

**Comment by torch2424** [comment]
@Heebe okay cool. Well I'm gonna be busy on stuff for the game all day. figure out special attacks with emanuel please <3

**Comment by Heebe** [comment]
@emanuelpartida

**Comment by torch2424** [comment]
Also, button config for this is contained in #112

**Comment by Heebe** [comment]
What do you mean by figure out special attacks?

**Comment by emanuelpartida** [comment]
instead of a fire ball i can make it like a fist? and earth spike you punch the ground is the goodish?

**Comment by torch2424** [comment]
@emanuelpartida I like the firball be a giant fireball fist. THat's be awesome, and how about instead of spikes, they are earth fists?

**Comment by torch2424** [comment]
@Heebe Like I said above "I'm being outvoted, and I think I'm just being too picky on this whole thing. How about this. lets just start over, I think I kinda influenced everything oddly. You and emanual can come up with an idea wether it be more buttons, more mana, orbs, specials drops or whatever. And then come back to me, and I'll code it up?"

However, you keep asking for my input. I would like for you an emanual, to sit down, from a clean slate decide what kind of special attakcs, power ups, or whatever you guys would like to do. Today is my dev day, and I'm focused on that

**Comment by Heebe** [comment]
I was including you because you are a part of this, anyway, Emauel and I can't sit down and talk today because he has work in like 2 hours.

**Comment by emanuelpartida** [comment]
i like your earth fist idea that sounds random and one of the power up could be a morph that turns punch guy to a giant demon that punches everything and another could be invincibility or super speed?  How about super flying fist like punch guy flies one direct and destroys his enemies with is mighty flying fist??

**Comment by Heebe** [comment]
We'll talk about those when I see you tomorrow @Emanuel, Starbucks?

**Comment by Heebe** [comment]
And @torch2424, I was just asking what you meant, that's all.

**Comment by emanuelpartida** [comment]
sure dude

**Comment by torch2424** [comment]
@Heebe I appreciate you including me. but yes that is what I mean <3

**Comment by Heebe** [comment]
Alright, Emanuel and I have agreed upon:

_Drops:_
There will be a total of five drops and they all will mostly be **UNCOMMON - RARE**.
1. +0.5x speed - stacks for up to a total of 2x overall speed, so cap will be hit after **2** consecutive pickups.
2. Refiil MP
_side note 1: MP will regen a bit slower than health, making this a very useful item in a tight spot if you're running out of MP in a jam._
3. +crit-rate - see side note 2
_side note 2: lower the crit-rate that's at the current level just slightly, so when you get the drop the crit-rate goes back up to how it is now or maybe slightly higher._
4. +att power - +25x attack power, caps at +.5x so cap will be hit after **2** consecutive pickups.
5. Demon Transformation - no dodge, 2x att power, health stays the same, player slows down, crit-chance +.25x (see side note 3)
_side note 3: this drop will be **VERY RARE**._

_side note 4: Drops 1, 3, 4, and 5 will work on a timer of 15 seconds of use, if the player happens to get an extra pickup of these the timer will be stacked for more extra time._
_side not 5: drops that have not been picked up will disappear after 40 seconds if not being picked up._

_Specials:_
Special will change after the level ranges 1 -10, 11 - 20, 21 - 30, etc. so that each special may be unique and the player would be able to use each kind of special we offer against a boss after each level range.
-Special's "level ranges" may very later, if specials still don't make it possible for a player to hit much higher levels.

:star:Which specials we use and at what level ranges are still being actively worked on but shall be worked out soon.:star:

**Comment by torch2424** [comment]
@Heebe Since you guys still have a little bit left to figure out, I'll be doing specials in the next alpha, but I'd like to have this decided by this milestone :D

**Comment by torch2424** [comment]
Hey Guys, if you could please let me know what levels have what specials, and give me all the art for this (Those that are missing), that'd be great, as I'd like to close this issue and move onto the next one!

**Comment by torch2424** [comment]
I'm going to move this into alpha 8.0, but I'd like to have this done ASAP, that way we can move into beta much sooner

**Comment by torch2424** [comment]
Also, one thing I'd like you guys to note #135 is something that needs to be done since we are using a special button and people's specials. This will take up a lot of time and resources. Compared to just drops which kevin could make. I will be not working on the game as much until after all of this is implemented, and we communicate more on the issues with needs discussion, as I can only code the content that I have. And I cannot go and create any new screens a trailer or anything until we ave finalized content
